mslib/msui/viewwindows.py

```diff
@@ -76,6 +76,13 @@ def closeEvent(self, event):
         Overloads QtGui.QMainWindow.closeEvent(). This method is called if
         Qt receives a window close request for our application window.
         """
+        if self.docks[5] is not None and self.mainwindow_signal_logout_mscolab is not None:
+            # ToDo instead of numbers [5] use names
+            # we need to disconnect the mainwindow_signal_logout when the topview is closed
+            # the next topview will connect one again
+            self.mainwindow_signal_logout_mscolab.disconnect()
```

Comment:
AFAIU this would disconnect _all_ slots from the `mainwindow.signal_logout_mscolab` signal, no matter where they were connected from. This is wrong, unless there is only ever one topview and nothing else connects to the signal. The widget should only disconnect the slots that it itself connected.

Reply:
I know, but not seen where this connect else is used.  And somehow the whole cleanup can be improved.

Manually this seem to work also with many topviews. 
weakref check may need a gc., currently I get many time a timeout and <[RuntimeError('wrapped C/C++ object of type MSUITopViewWindow has been deleted') raised in repr()] that needs something better too.

Reply:
> I know, but not seen where this connect else is used.

Both the MSUITopViewWindow and the MultipleFlightpathControlWidget have a connect call on this signal AFAICT.

Even if it isn't a problem _right now_, it can become one at any time when a different widget decides to listen on that signal as well. It will be very hard to debug then. Also, it is just a violation of the separation of the widgets that is avoidable, i.e. it is easy enough to track the result of the connect call and specifically just disconnect that one (or maybe that is not even necessary if the slot wasn't a lambda but a method of the object, since Qt does a check if the slot still exists before calling the slot, AFAIU).

Reply:
the lambda was the culprit. I am a bit surprised about the QT magic, but seems to work at least manually.